lib/src/main/java/uk/gov/di/data/lep/library/services/PartialDateDeserialiser.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.di.data.lep.library.services;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.temporal.TemporalAccessor;
+
+public class PartialDateDeserialiser extends JsonDeserializer<TemporalAccessor> {
+    @Override
+    public TemporalAccessor deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+        var token = jsonParser.getCurrentToken();
+
+        if (token.equals(JsonToken.VALUE_STRING)) {
+            var text = jsonParser.getText().trim();
+            var countOfDashes = text.split("-", -1).length - 1;
```

Comment:
Can we do the split once, then count "number of date parts" instead? (that saves us doing the same split again on ln 26)